src/server/database/Database/Transaction.cpp

```diff
@@ -67,16 +69,26 @@ bool TransactionTask::Execute()
 {
     int errorCode = m_conn->ExecuteTransaction(m_trans);
     if (!errorCode)
+    {
+        m_trans->Cleanup();
         return true;
+    }
 
     if (errorCode == ER_LOCK_DEADLOCK)
     {
         // Make sure only 1 async thread retries a transaction so they don't keep dead-locking each other
-        std::lock_guard<std::mutex> lock(_deadlockLock);
+        //std::lock_guard<std::mutex> lock(_deadlockLock);
```

Comment:
why did you comment this line ?

Reply:
Ariel said: "static analysis tools will complain for sleeping (and with reason!) with a locked mutex" so I commented it out.

Reply:
It doesn't work like that, you don't edit the code to make static analysis warnings disappear. This is a very delicate part of the core that requires a lot of attention so every modified line requires to fully understand what's happening.

The mutex is there to make sure 2 different threads deadlocking on the same SQL rows will not keep trying again at the same time to submit the transaction, resulting again in a deadlock.

I would change the loop counter to a timed loop that keeps trying for a fixed amount of time instead without the thread_sleep